### PR TITLE
pallet Collective: fix genesis member sort order

### PR DESCRIPTION
**Pull request by liamaharon**
The Collective pallet currently does not check that genesis members are sorted or sort them before setting them as expected by the pallet:

https://github.com/paritytech/substrate/blob/9370f42ab35f7013f70b703884c9884888a3b153/frame/collective/src/lib.rs#L281-L285

`kitchen-sink-runtime` invariants are currently broken due to this bug.

This PR adds sorting logic into the `initialize_members`, ensuring genesis members are inserted in sorted order.

Props to @Szegoo for adding the Collective try-state hooks (https://github.com/paritytech/substrate/pull/13645) that caught this!

--- 

Open question: what's the likelihood there're other pallets that suffer from a similar bug? Should an issue need to be opened to investigate that?

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

File: frame/collective/src/lib.rs [hunk 1]
```diff
@@ -1107,6 +1107,8 @@ impl<T: Config<I>, I: 'static> InitializeMembers<T::AccountId> for Pallet<T, I>
 	fn initialize_members(members: &[T::AccountId]) {
 		if !members.is_empty() {
 			assert!(<Members<T, I>>::get().is_empty(), "Members are already initialized!");
+			let mut members = members.to_vec();
+			members.sort();
```
Comment (ggwpez) [inline frame/collective/src/lib.rs:1111]: I guess duplicates are fine?  
Please add a duplicate to the test to check it.

**Review by ggwpez — APPROVED** [review]

**Comment by ggwpez** [comment]
>Open question: what's the likelihood there're other pallets that suffer from a similar bug? Should an issue need to be opened to investigate that?

Adding more and more `try-state` hooks should be enough IMHO.

File: frame/collective/src/lib.rs (hunk 1 above)
Comment (muharem) [reply frame/collective/src/lib.rs:1111]: I think we should also assert the `MaxMembers` limit here.
I wonder why the `Vec` is not bounded for the `Members` storage. we should probably migrate to bounded?

Comment (liamaharon) [reply frame/collective/src/lib.rs:1115]: Duplicates are already tested.

MaxMembers was not - good suggestion @muharem

**Pull request merged**